src/lib.rs

```diff
@@ -36,6 +36,90 @@
 //! This crate provides utilities that make writing both clients and servers easier. You'll find
 //! the client helper as `Client` in the protocol module you're working with (e.g.,
 //! [`pipeline::Client`]), and the server helper as `Server` in the same place.
+//!
+//! # Example
+//! ```rust
+//! # type StdError = Box<dyn std::error::Error + Send + Sync + 'static>;
+//! # use std::pin::Pin;
+//! # use std::boxed::Box;
+//! # use tokio::sync::mpsc;
+//! # use tokio::io::{AsyncWrite, AsyncRead};
+//! # use futures_core::task::{Context, Poll};
+//! # use futures_util::{never::Never, future::{poll_fn, ready, Ready}};
+//! # use tokio_tower::pipeline;
+//! /// Wrapper around our transport.
+//! /// Need to implement Sink and Stream for this.
```

Comment:
A more elaborate comment for what this type is and why it's needed would probably be good.